en/material-system/effect-2.x-to-3.0.md

```diff
@@ -0,0 +1,254 @@
+# Cocos Creator 3.0 Material Upgrade Guide
+
+> This article will detail the considerations for upgrading Cocos Creator 2.x materials to v3.0.
+
+## 1. Introduction to the base design of the material system
+
+### 1.1 The material system framework of Cocos Creator
+
+The material system consists of four core classes from top to bottom, Material, Effect, Technique and Pass, whose relationship can be understood by the following class diagram:
+
+![effect](material.png)
+
+#### Material
+
+The Material asset can be seen as an asset instance of the EffectAsset in the scene, and its own configurable parameters are effectAsset, technique, defines, states.
+
+#### Effect
+
+An Effect asset represents a material type and is the most important core asset in the material system. If you want to implement a custom shading effect in the engine, you need to write a custom Effect.
+
+#### Technique
+
+We refer to the solution to accomplish a final effect as a rendering technique, and a technique can be fused by one or more passes.
+
+#### Pass
+
+A Pass is a single GPU draw, typically including a vertex shader and a slice shader, and there are many optional configuration parameters for Pass in Creator.
+
+### 1.2 Material Instance Panel
+
+The Material instance panel is the most intuitive material editing window for all developers, and all actual material instances are configured through it.
+
+The Material instance panel in Cocos Creator 3.0 is as follows:
+
+![effect](material3.png)
+
+The Material instance panel of Cocos Creator 2.x looks like this
+
+![effect](material10.jpg)
+
+As you can see from the above two figures, the instance panel in v3.0 is quite complex compared to v2.4, partly because of the increased complexity of the material configuration, and partly because of the enhanced functionality of the panel.
```

Comment:
```suggestion
Notice from the above two figures, the instance panel in v3.0 is quite complex compared to v2.4, partly because of the increased complexity of the material configuration, and partly because of the enhanced functionality of the panel.
```